### ISSUE TITLE
libsemigroups instructions printed by configure are not sufficient

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/Semigroups#420]
I just got this:
```
...
checking for src/libsemigroups/src/libsemigroups-config.h... no
configure: error: libsemigroups-config.h is required, either download the
latest release version from

    https://github.com/james-d-mitchell/libsemigroups

or do

    ./autogen.sh ; ./configure

in the src/libsemigroups directory
```

Following those instructions did not help; I also had to `make`. Moreover, I suggest using `&&` instead of `;` for your own benefit, when/if somebody files a bug report based on this: if they copy&paste the line with `;` and it runs in an error, then past experience says they'll tell you the last command failed, because that's what they see; but in fact `autogen.sh` may have failed, and that failure can be lost among print outs by an old `configure` version. With `&&` the first command that fails prevents the later commands from being executed.

I.e. I'd use this:

   ./autogen.sh && ./configure && make

**Comment by fingolfin** [comment]
Actually, I partially take that back: not even doing `make` helps, I still get that error.

I then noticed that `src/libsemigroups` is a git repository and was several versions behind, so I did a `git pull` in there and tried again. Still no luck; but at least doing `make` in there now printed:
```
 make
  CXX      src/libsemigroups_la-blocks.lo
In file included from src/blocks.cc:30:
In file included from src/blocks.h:29:
src/libsemigroups-debug.h:26:10: fatal error: 'libsemigroups-config.h' file not found
#include "libsemigroups-config.h"
         ^
1 error generated.
make: *** [Makefile:658: src/libsemigroups_la-blocks.lo] Error 1
```

So I finally did a `git clean -f -x` in `libsemigroups` and tried yet again, still no luck

**Comment by james-d-mitchell** [comment]
Thanks for this @fingolfin I didn't notice this until today, sorry for the delay. 

The corrected instructions printed by `./configure` for Semigroups say to do `./autogen.sh && ./configure` inside the `src/libsemigroups` directory, this is correct, `./configure` for libsemigroups creates the `libsemigroups-config.h` file, and this is required for the Semigroups package in GAP. You do not need to `make` the library `libsemigroups` at all, only run `./autogen.sh && ./configure` in `src/libsemigroups` and then build the Semigroups package for GAP in the usual way.

**Comment by fingolfin** [comment]
Found the problem (I think): I did a `make clean`, too, and that deletes the config header:
```
clean-local: clean-local-check
	rm -f src/libsemigroups-config.h
```
But there is no rule to regenerate it. Result:
```
$ make clean
 rm -f lstest
test -z "libsemigroups.la" || rm -f libsemigroups.la
rm -f ./so_locations
rm -rf .libs _libs
rm -rf src/.libs src/_libs
rm -rf tests/log lstest
rm -rf benchmark/src/build
rm -f benchmark/gbench/*.pyc
rm -f benchmark/gbench/__py_cache__/*.pyc
rm -f src/libsemigroups-config.h
rm -f *.o
rm -f src/*.o
rm -f src/*.lo
rm -f tests/*.o
rm -f *.lo
$ make
  CXX      src/libsemigroups_la-blocks.lo
In file included from src/blocks.cc:30:
In file included from src/blocks.h:29:
src/libsemigroups-debug.h:26:10: fatal error: 'libsemigroups-config.h' file not found
#include "libsemigroups-config.h"
         ^
1 error generated.
make: *** [Makefile:662: src/libsemigroups_la-blocks.lo] Error 1
```

**Comment by james-d-mitchell** [comment]
Ah, right, this ought to be under `distclean-local` not `clean-local`

**Comment by james-d-mitchell** [comment]
But @fingolfin that's still in libsemigroups, and not Semigroups itself.

**Comment by james-d-mitchell** [comment]
I think I have resolved this with PR #427, any further comments or concerns @fingolfin?

**Comment by fingolfin** [comment]
None at this time. Should I run into any problems again, I'll just open a fresh issue.
